tests/test_it_deploy.py

```diff
@@ -2,23 +2,65 @@
 from fastapi.testclient import TestClient
 from hayhooks.server import app
 from pathlib import Path
+from hayhooks.server.pipelines.registry import registry
 
 client = TestClient(app)
 
+
+@pytest.fixture(autouse=True)
+def clear_registry():
+    registry.clear()
+
+
 # Load pipeline definitions from test_files
 test_files = Path(__file__).parent / "test_files" / "working_pipelines"
 pipeline_data = [{"name": file.stem, "source_code": file.read_text()} for file in test_files.glob("*.yml")]
 
 
-@pytest.mark.parametrize("pipeline_data", pipeline_data)
-def test_deploy_pipeline_def(pipeline_data: dict):
+def deploy_pipeline(pipeline_data: dict):
     deploy_response = client.post(
         "/deploy", json={"name": pipeline_data["name"], "source_code": pipeline_data["source_code"]}
     )
+
+    return deploy_response
```

Comment:
I see that this function and the following one (`undeploy_pipeline`) are used in several test modules.

I thought that importing functions from other test modules was a bad practice, but it turns out that it is a permitted but controversial practice ([my search on Perplexity](https://www.perplexity.ai/search/in-pytest-is-importing-a-test-JPgOVR6kRCqWg4BcvCqMRg) :smiley: ).

I would transform these functions into fixtures and put them in tests/conftest.py, so that we can reuse them.
WDYT?

Reply:
That's a good point actually! I've did a refactoring with also other methods (deploy, undeploy, draw, status) and used them as fixures in tests